backend/prisma/schema.prisma

```diff
@@ -40,20 +45,49 @@ model GroupMember {
   addedById Int
   addedAt   DateTime  @default(now())
   group     CodeGroup @relation("GroupMembers", fields: [groupId], references: [id])
-  user      User      @relation("UserGroupMemberships", fields: [userId], references: [id])
-  addedBy   User      @relation("AddedByUser", fields: [addedById], references: [id])
+  user      User      @relation("GroupsAddedTo", fields: [userId], references: [id])
 
   @@unique([groupId, userId])
 }
 
 model File {
   id            Int       @id @default(autoincrement())
   fileName      String
-  fileContent   String
+  fileContent   String?
   creatorId     Int
   groupId       Int
   createdAt     DateTime  @default(now())
   lastUpdatedAt DateTime  @updatedAt
-  user          User      @relation("UserFiles", fields: [creatorId], references: [id])
+  creator       User      @relation("UserFiles", fields: [creatorId], references: [id])
   codeGroup     CodeGroup @relation("GroupFiles", fields: [groupId], references: [id])
 }
+
+model Skill {
+  id          Int        @id @default(autoincrement())
+  skill       String
+  user        User?      @relation(fields: [userId], references: [id])
+  userId      Int?
+  CodeGroup   CodeGroup? @relation(fields: [codeGroupId], references: [id])
+  codeGroupId Int?
+}
+
+model Interest {
+  id       Int    @id @default(autoincrement())
+  Interest String
+  User     User?  @relation(fields: [userId], references: [id], onDelete: Cascade)
+  userId   Int?
+}
+
+model Notification {
+  id          Int      @id @default(autoincrement())
+  message     String
+  category    String
+  isImportant Boolean
```

Comment:
How will this field be used?

Reply:
To filter between important and non-important notifications

Reply:
Yeah I mean is this a stretch feature you plan to add? It's not used currently right

Reply:
Yes, it is not currently there; I plan to add it. Do you want me to remove the field till when I am ready to implement that?

Reply:
If you think you'll have time to come back to it you can leave it.

Reply:
Okay, I will remove it now. I can always add it after, if I have the time.